workloads/kube-burner/common.sh

```diff
@@ -242,3 +246,70 @@ prep_networkpolicy_workload() {
   oc apply -f workloads/networkpolicy/clusterrole.yml
   oc apply -f workloads/networkpolicy/clusterrolebinding.yml
 }
+
+churn() {
+  log "Starting to churn workload"
+
+  churn_start=`date +%s`
+  churn_end_time=$((${churn_start} + ${CHURN_DURATION}*60))
+
+  log "Churn duration: ${CHURN_DURATION} minutes"
+  log "Churn wait duration: ${CHURN_WAIT} seconds"
+  log "Churn percentage: ${CHURN_PERCENT}%"
+  log "Churn type: ${CHURN_TYPE}"
+
+  namespace_array=(`kubectl get namespaces -l kube-burner-uuid=${UUID} --no-headers | awk '{print $1}'`)
+  namespace_count=${#namespace_array[@]}
+  
+  # The number of iterations we need to modify per round of churn (min 1)
+  modify_count=$((JOB_ITERATIONS*CHURN_PERCENT/100))
+  if [[ ${modify_count} -eq 0 ]]; then modify_count=1; fi
+  
+  current_time=`date +%s` 
+  while [ ${current_time} -le ${churn_end_time} ]; do
+    for ((i=0; i<${modify_count}; i++)); do
+      #Pick random Namespace from the list
+      rand_ns=$(($RANDOM%${namespace_count}))
+
+      if [[ ${CHURN_TYPE} == "pod" ]]; then
+        rep_sets=(`kubectl get replicasets -n ${namespace_array[$rand_ns]} --no-headers | awk '{print $1}'`)
+        #Pick a random replicaset from the namespace
+        rand_rep=$(($RANDOM%${#rep_sets[@]}))
+        #set to 0 replicas; it will automatically get swapped back to the appropriate number of replicas
+        kubectl scale --replicas 0 replicaset ${rep_sets[$rand_rep]} -n ${namespace_array[$rand_ns]} > /dev/null &
```

Comment:
I think all the logic of the pod churn_type could be simplified with 
`oc delete pod --all -n ${namespace_array[$rand_ns]}`
 
These pods will be automatically re-created by their corresponding controllers

Reply:
True, however deleting all the pods in the namespace would be likely more than what I have coded in at the moment. That said I am fine moving to this as an option.

Reply:
You mean that more lines would be required?, My suggestion is to replace all that block with just one line

Reply:
Nope. I mean that the current implementation does a random replica set at a time and theres usually multiple of those per namespace in our cluster-density tests. You delete ALL the pods in a namespace while mines potentially fewer. Again, its splitting hairs and I'm fine deleting all the pods.

Reply:
Updated to use kubectl delete pod --all....